src/qz/printer/status/CupsStatusHandler.java

```diff
@@ -52,26 +55,42 @@ private void getNotifications() {
                 String jobState = Cups.INSTANCE.ippEnumString("job-state", Cups.INSTANCE.ippGetInteger(jobStateAttr, 0));
                 String jobName = cups.ippGetString(jobNameAttr, 0, "");
 
+                // todo: this should be job url maybe
+                ArrayList<Status> oldStatuses = lastJobStatusMap.getOrDefault(printer + jobId, new ArrayList<>());
+                ArrayList<Status> newStatuses = new ArrayList<>();
+
+                boolean completed = false;
                 int attrCount = cups.ippGetCount(jobStateReasonsAttr);
                 for (int i = 0;  i < attrCount; i++) {
                     String reason = cups.ippGetString(jobStateReasonsAttr, i, "");
-                    statuses.add(NativeStatus.fromCupsJobStatus(reason, jobState, printer, jobId, jobName));
+                    Status status = NativeStatus.fromCupsJobStatus(reason, jobState, printer, jobId, jobName);
+                    if (!oldStatuses.contains(status)) statuses.add(status);
+                    if (status.getCode() == NativeJobStatus.COMPLETE) completed = true;
+                }
+                if (completed) {
+                    lastJobStatusMap.remove(printer + jobId);
+                } else {
+                    lastJobStatusMap.put(printer + jobId, newStatuses);
```

Comment:
Do you ever add anything to `newStatuses` ?

Reply:
good catch